src/datasets/data_files.py

```diff
@@ -296,10 +296,13 @@ def get_patterns_locally(base_path: str) -> Dict[str, List[str]]:
 def _resolve_single_pattern_in_dataset_repository(
     dataset_info: huggingface_hub.hf_api.DatasetInfo,
     pattern: str,
+    base_path: Optional[str] = None,
     allowed_extensions: Optional[list] = None,
 ) -> List[PurePath]:
     data_files_ignore = FILES_TO_IGNORE
     fs = HfFileSystem(repo_info=dataset_info)
+    if base_path:
+        pattern = os.path.join(base_path, pattern)
```

Comment:
The `HfFileSystem` expects paths with a `/` separator. However `os.path.join` on windows uses the `\` separator.

```suggestion
        pattern = f"{base_path}/{pattern}"
```